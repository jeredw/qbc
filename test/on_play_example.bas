    ON PLAY(3) GOSUB Background
    PLAY ON
    Music$ = "MBo3L8ED+ED+Eo2Bo3DCL2o2A"
    PLAY Music$
    LOCATE 2, 1: PRINT "Press any key to stop.";
    DO WHILE INKEY$ = "": LOOP
    END

    Background:
        i% = i% + 1
        LOCATE 1, 1: PRINT "Background called "; i%; "time(s)";
        PLAY Music$
        RETURN
