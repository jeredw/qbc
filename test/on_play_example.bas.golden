(program 
  (statement 
    (on_event_gosub_statement ON PLAY 
      ( 
        (expr 
          (literal 3)) ) GOSUB 
      (target 
        (text_label Background)))) \n 
  (statement 
    (event_control_statement PLAY ON)) \n 
  (statement 
    (assignment_statement 
      (typed_id Music $) = 
      (expr 
        (literal "MBo3L8ED+ED+Eo2Bo3DCL2o2A")))) \n 
  (statement 
    (play_statement PLAY 
      (expr 
        (typed_id Music $)))) \n 
  (statement 
    (call_statement LOCATE 
      (call_argument_list 
        (call_argument 
          (expr 
            (literal 2))) , 
        (call_argument 
          (expr 
            (literal 1)))))) : 
  (statement 
    (print_statement PRINT 
      (print_args 
        (expr 
          (literal "Press any key to stop."))) ;)) \n 
  (statement 
    (do_loop_statement DO 
      (do_condition WHILE 
        (expr 
          (expr 
            (typed_id INKEY $)) = 
          (expr 
            (literal "")))) 
      (block :) LOOP)) \n 
  (statement 
    (end_statement END)) \n statement \n 
  (label 
    (text_label Background) :) statement \n 
  (statement 
    (assignment_statement 
      (typed_id i %) = 
      (expr 
        (expr 
          (typed_id i %)) + 
        (expr 
          (literal 1))))) \n 
  (statement 
    (call_statement LOCATE 
      (call_argument_list 
        (call_argument 
          (expr 
            (literal 1))) , 
        (call_argument 
          (expr 
            (literal 1)))))) : 
  (statement 
    (print_statement PRINT 
      (print_args 
        (expr 
          (literal "Background called ")) ; 
        (print_args 
          (expr 
            (typed_id i %)) ; 
          (print_args 
            (expr 
              (literal "time
                (s)"))))) ;)) \n 
  (statement 
    (play_statement PLAY 
      (expr 
        (typed_id Music $)))) \n 
  (statement 
    (return_statement RETURN)) \n)
